lib/matplotlib/ticker.py

```diff
@@ -1016,33 +1039,29 @@ def __call__(self, x, pos=None):
         else:
             base = '%s' % b
 
-        if coeff in self.sublabel:
-            if not is_decade and self.labelOnlyBase:
-                return ''
-            elif not is_decade:
-                return self._non_decade_format(sign_string, base, fx, usetex)
-            else:
-                if usetex:
-                    return (r'$%s%s^{%d}$') % (sign_string,
-                                               base,
-                                               nearest_long(fx))
-                else:
-                    return ('$%s$' % _mathdefault(
-                        '%s%s^{%d}' %
-                        (sign_string, base, nearest_long(fx))))
-        else:
+        if self.sublabel_filtering and coeff not in self._sublabels:
+            return ''
+        if not is_decade and self.labelOnlyBase:
             return ''
 
+        if not is_decade:
```

Comment:
Is there an interest in not "factorizing" this `if statement` with the previous one (l. 1044)? Something like
```python
if not is_decade:
    if self.labelOnlyBase:
        return ''
    else:
        return self._non_decade_format(sign_string, base, fx, usetex)
else:
    ...
```